### ISSUE TITLE
Sporadic NoClassDefFoundError for SimplePolicyConfiguration on app deployment (related to Exousia)

### ISSUE THREAD

**Issue by OndroMih** [eclipse-ee4j/glassfish#24543]
## Environment Details
* GlassFish Version: 7.0.7 (I noticed this also in a few previous versions but didn't raise an issue before)
* JDK version: Java 17
* OS: Linux
* Database: N/A

----------

## Problem Description

Sometimes, very rarely actually, when I deploy an application to a freshly started GlassFish server, the app doesn't deploy with `java.lang.NoClassDefFoundError: Could not initialize class org.glassfish.exousia.modules.locked.SimplePolicyConfiguration` in server log. This keeps happening when I redeploy the application until I restart the server. When I start GlassFish again, it usually gets fixed.

## Steps to reproduce

The problem happens very rarely, it looks like it's some race condition. So it's not easy to reproduce. I don't know if there's any specific functionality in the app being deployed or it can happen with any app. I also saw this exception when I ran the GlassFish test suite - some IT tests failed because of this exception because they couldn't deploy the test app.

Here's the stacktrace (truncated to contain the trace since the `DeployCommand` is executed):

```
java.lang.NoClassDefFoundError: Could not initialize class org.glassfish.exousia.modules.locked.SimplePolicyConfiguration
	at org.glassfish.exousia.modules.locked.SimplePolicyProvider.getPermissions(SimplePolicyProvider.java:106)
	at java.base/java.security.Policy.initPolicy(Policy.java:347)
	at java.base/java.security.Policy.setPolicy(Policy.java:289)
	at com.sun.enterprise.security.PolicyLoader.loadPolicy(PolicyLoader.java:138)
	at com.sun.enterprise.security.SecurityLifecycle.onInitialization(SecurityLifecycle.java:127)
	at com.sun.enterprise.security.SecurityLifecycle.postConstruct(SecurityLifecycle.java:153)
	at org.jvnet.hk2.internal.ClazzCreator.postConstructMe(ClazzCreator.java:356)
	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:410)
	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:466)
	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:59)
	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:47)
	at org.glassfish.hk2.utilities.cache.Cache$OriginThreadAwareFuture$1.call(Cache.java:74)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.glassfish.hk2.utilities.cache.Cache$OriginThreadAwareFuture.run(Cache.java:131)
	at org.glassfish.hk2.utilities.cache.Cache.compute(Cache.java:176)
	at org.jvnet.hk2.internal.SingletonContext.findOrCreate(SingletonContext.java:98)
	at org.jvnet.hk2.internal.Utilities.createService(Utilities.java:2109)
	at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:93)
	at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:67)
	at com.sun.enterprise.security.ee.SecuritySniffer.setup(SecuritySniffer.java:106)
	at com.sun.enterprise.v3.server.ContainerStarter.startContainer(ContainerStarter.java:69)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.setupContainer(ApplicationLifecycle.java:1050)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.setupContainerInfos(ApplicationLifecycle.java:768)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:413)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:259)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:467)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:570)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:566)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:376)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:565)
...
```

## Impact of Issue
[//]: # "How has this issue affected you? What are you trying to accomplish?"

Stability of the product. User experience. A thing that should work all the time (start server and deploy app) sometimes doesn't work and requres restarting the server.

[//]: # "Please choose one each Type, Component, and Priority label"
Type: bug
Component: Exousia, Deployer, Classloading
Priority: high

**Comment by OndroMih** [comment]
Hi, @arjantijms , do you know what might cause this?

**Comment by pzygielo** [comment]
I've just seen similar with glassfish-7.0.25:

```
[2026-04-12T22:02:40.824002Z] [GF 7.0.25] [SEVERE] [] [jakarta.enterprise.system.core] [tid: _ThreadID=50 _ThreadName=admin-listener(4)] [levelValue: 1000] [[
  Exception while preparing the app : A MultiException has 2 exceptions.  They are:
1. java.lang.NoClassDefFoundError: Could not initialize class org.glassfish.exousia.modules.locked.SimplePolicyConfiguration
2. java.lang.IllegalStateException: Unable to perform operation: post construct on com.sun.enterprise.security.SecurityLifecycle

MultiException stack 1 of 2
java.lang.NoClassDefFoundError: Could not initialize class org.glassfish.exousia.modules.locked.SimplePolicyConfiguration
	at org.glassfish.exousia.modules.locked.SimplePolicyProvider.getPermissions(SimplePolicyProvider.java:121)
	at java.base/java.security.Policy.initPolicy(Policy.java:347)
	at java.base/java.security.Policy.setPolicy(Policy.java:289)
	at com.sun.enterprise.security.PolicyLoader.loadPolicy(PolicyLoader.java:137)
	at com.sun.enterprise.security.SecurityLifecycle.onInitialization(SecurityLifecycle.java:127)
	at com.sun.enterprise.security.SecurityLifecycle.postConstruct(SecurityLifecycle.java:153)
	at org.jvnet.hk2.internal.ClazzCreator.postConstructMe(ClazzCreator.java:356)
	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:410)
	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:479)
	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:61)
	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:49)
	at org.glassfish.hk2.utilities.cache.Cache$OriginThreadAwareFuture$1.call(Cache.java:74)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.glassfish.hk2.utilities.cache.Cache$OriginThreadAwareFuture.run(Cache.java:131)
	at org.glassfish.hk2.utilities.cache.Cache.compute(Cache.java:176)
	at org.jvnet.hk2.internal.SingletonContext.findOrCreate(SingletonContext.java:100)
	at org.jvnet.hk2.internal.Utilities.createService(Utilities.java:2111)
	at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:98)
	at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:68)
	at com.sun.enterprise.security.ee.SecuritySniffer.setup(SecuritySniffer.java:106)
	at com.sun.enterprise.v3.server.ContainerStarter.startContainer(ContainerStarter.java:69)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.setupContainer(ApplicationLifecycle.java:1039)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.setupContainerInfos(ApplicationLifecycle.java:758)
	at org.glassfish.javaee.full.deployment.EarDeployer.prepareBundle(EarDeployer.java:274)
	at org.glassfish.javaee.full.deployment.EarDeployer$1.doBundle(EarDeployer.java:143)
	at org.glassfish.javaee.full.deployment.EarDeployer$1.doBundle(EarDeployer.java:139)
	at org.glassfish.javaee.full.deployment.EarDeployer.doOnBundles(EarDeployer.java:210)
	at org.glassfish.javaee.full.deployment.EarDeployer.doOnAllTypedBundles(EarDeployer.java:218)
	at org.glassfish.javaee.full.deployment.EarDeployer.doOnAllBundles(EarDeployer.java:243)
	at org.glassfish.javaee.full.deployment.EarDeployer.prepare(EarDeployer.java:139)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.prepareModule(ApplicationLifecycle.java:978)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:466)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:256)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:474)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:570)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:566)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:376)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:565)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:596)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:588)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:376)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:587)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1480)
	at com.sun.enterprise.v3.admin.CommandRunnerExecutionContext.execute(CommandRunnerExecutionContext.java:251)
	at com.sun.enterprise.v3.admin.CommandRunnerExecutionContext.execute(CommandRunnerExecutionContext.java:134)
	at org.glassfish.admin.rest.resources.admin.CommandResource.executeCommand(CommandResource.java:382)
	at org.glassfish.admin.rest.resources.admin.CommandResource.execCommandSimpInMultOut(CommandResource.java:225)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:146)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:189)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:93)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:274)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:266)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:253)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:696)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:367)
	at org.glassfish.admin.rest.adapter.JerseyContainerCommandService$1.service(JerseyContainerCommandService.java:149)
	at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:159)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:429)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:143)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:174)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:153)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:196)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:88)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:246)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:178)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:118)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:96)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:51)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:510)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:82)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:83)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:101)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:535)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:515)
	at java.base/java.lang.Thread.run(Thread.java:840)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.IllegalStateException: RoleMapper registration failed! [in thread "FelixStartLevel"]
	at org.glassfish.exousia.modules.locked.SimplePolicyConfiguration.<clinit>(SimplePolicyConfiguration.java:109)
	at org.glassfish.exousia.modules.locked.SimplePolicyProvider.doImplies(SimplePolicyProvider.java:159)
	at org.glassfish.exousia.modules.locked.SimplePolicyProvider.implies(SimplePolicyProvider.java:150)
	at java.base/java.security.ProtectionDomain.implies(ProtectionDomain.java:325)
	at java.base/java.security.ProtectionDomain.impliesWithAltFilePerm(ProtectionDomain.java:357)
	at java.base/java.security.AccessControlContext.checkPermission(AccessControlContext.java:463)
	at java.base/java.security.AccessController.checkPermission(AccessController.java:1068)
	at org.glassfish.security.services.common.SecurityAccessValidator.checkPerm(SecurityAccessValidator.java:256)
	at org.glassfish.security.services.common.SecurityAccessValidator.validateLookup(SecurityAccessValidator.java:164)
	at org.glassfish.security.services.common.SecurityAccessValidator.validate(SecurityAccessValidator.java:59)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.callValidate(ServiceLocatorImpl.java:230)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.validate(ServiceLocatorImpl.java:274)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.getDescriptors(ServiceLocatorImpl.java:346)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.getDescriptors(ServiceLocatorImpl.java:390)
	at org.glassfish.hk2.extras.events.internal.DefaultTopicDistributionService.configurationChanged(DefaultTopicDistributionService.java:539)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.callAllConfigurationListeners(ServiceLocatorImpl.java:2124)
	at org.jvnet.hk2.internal.ServiceLocatorImpl.addConfiguration(ServiceLocatorImpl.java:2191)
	at org.jvnet.hk2.internal.DynamicConfigurationImpl.commit(DynamicConfigurationImpl.java:265)
	at org.jvnet.hk2.osgiadapter.HK2Main$HK2ServiceTrackerCustomizer.addingService(HK2Main.java:368)
	at OSGi//org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:944)
	at OSGi//org.osgi.util.tracker.ServiceTracker$Tracked.customizerAdding(ServiceTracker.java:872)
	at OSGi//org.osgi.util.tracker.AbstractTracked.trackAdding(AbstractTracked.java:256)
	at OSGi//org.osgi.util.tracker.AbstractTracked.track(AbstractTracked.java:229)
	at OSGi//org.osgi.util.tracker.ServiceTracker$Tracked.serviceChanged(ServiceTracker.java:904)
	at OSGi//org.apache.felix.framework.EventDispatcher.invokeServiceListenerCallback(EventDispatcher.java:990)
	at OSGi//org.apache.felix.framework.EventDispatcher.fireEventImmediately(EventDispatcher.java:838)
	at OSGi//org.apache.felix.framework.EventDispatcher.fireServiceEvent(EventDispatcher.java:545)
	at OSGi//org.apache.felix.framework.Felix.fireServiceEvent(Felix.java:4863)
	at OSGi//org.apache.felix.framework.Felix.registerService(Felix.java:3834)
	at OSGi//org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:328)
	at OSGi//org.apache.felix.framework.BundleContextImpl.registerService(BundleContextImpl.java:501)
	at org.apache.felix.cm.impl.ConfigurationManager.start(ConfigurationManager.java:199)
	at org.apache.felix.cm.impl.ConfigurationAdminStarter.activate(ConfigurationAdminStarter.java:78)
	at org.apache.felix.cm.impl.DependencyTracker.<init>(DependencyTracker.java:92)
	at org.apache.felix.cm.impl.Activator.start(Activator.java:114)
	at OSGi//org.apache.felix.framework.util.SecureAction.startActivator(SecureAction.java:849)
	at OSGi//org.apache.felix.framework.Felix.activateBundle(Felix.java:2429)
	at OSGi//org.apache.felix.framework.Felix.startBundle(Felix.java:2335)
	at OSGi//org.apache.felix.framework.Felix.setActiveStartLevel(Felix.java:1566)
	at OSGi//org.apache.felix.framework.FrameworkStartLevelImpl.run(FrameworkStartLevelImpl.java:297)
	... 1 more
MultiException stack 2 of 2
java.lang.IllegalStateException: Unable to perform operation: post construct on com.sun.enterprise.security.SecurityLifecycle
	at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:428)
	at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:479)
	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:61)
	at org.jvnet.hk2.internal.SingletonContext$1.compute(SingletonContext.java:49)
	at org.glassfish.hk2.utilities.cache.Cache$OriginThreadAwareFuture$1.call(Cache.java:74)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at org.glassfish.hk2.utilities.cache.Cache$OriginThreadAwareFuture.run(Cache.java:131)
	at org.glassfish.hk2.utilities.cache.Cache.compute(Cache.java:176)
	at org.jvnet.hk2.internal.SingletonContext.findOrCreate(SingletonContext.java:100)
	at org.jvnet.hk2.internal.Utilities.createService(Utilities.java:2111)
	at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:98)
	at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:68)
	at com.sun.enterprise.security.ee.SecuritySniffer.setup(SecuritySniffer.java:106)
	at com.sun.enterprise.v3.server.ContainerStarter.startContainer(ContainerStarter.java:69)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.setupContainer(ApplicationLifecycle.java:1039)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.setupContainerInfos(ApplicationLifecycle.java:758)
	at org.glassfish.javaee.full.deployment.EarDeployer.prepareBundle(EarDeployer.java:274)
	at org.glassfish.javaee.full.deployment.EarDeployer$1.doBundle(EarDeployer.java:143)
	at org.glassfish.javaee.full.deployment.EarDeployer$1.doBundle(EarDeployer.java:139)
	at org.glassfish.javaee.full.deployment.EarDeployer.doOnBundles(EarDeployer.java:210)
	at org.glassfish.javaee.full.deployment.EarDeployer.doOnAllTypedBundles(EarDeployer.java:218)
	at org.glassfish.javaee.full.deployment.EarDeployer.doOnAllBundles(EarDeployer.java:243)
	at org.glassfish.javaee.full.deployment.EarDeployer.prepare(EarDeployer.java:139)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.prepareModule(ApplicationLifecycle.java:978)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:466)
	at com.sun.enterprise.v3.server.ApplicationLifecycle.deploy(ApplicationLifecycle.java:256)
	at org.glassfish.deployment.admin.DeployCommand.execute(DeployCommand.java:474)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:570)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2$1.run(CommandRunnerImpl.java:566)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:376)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$2.execute(CommandRunnerImpl.java:565)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:596)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl$3.run(CommandRunnerImpl.java:588)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/javax.security.auth.Subject.doAs(Subject.java:376)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:587)
	at com.sun.enterprise.v3.admin.CommandRunnerImpl.doCommand(CommandRunnerImpl.java:1480)
	at com.sun.enterprise.v3.admin.CommandRunnerExecutionContext.execute(CommandRunnerExecutionContext.java:251)
	at com.sun.enterprise.v3.admin.CommandRunnerExecutionContext.execute(CommandRunnerExecutionContext.java:134)
	at org.glassfish.admin.rest.resources.admin.CommandResource.executeCommand(CommandResource.java:382)
	at org.glassfish.admin.rest.resources.admin.CommandResource.execCommandSimpInMultOut(CommandResource.java:225)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:146)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:189)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$ResponseOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:176)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:93)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:478)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:400)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:274)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:266)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:253)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:696)
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:367)
	at org.glassfish.admin.rest.adapter.JerseyContainerCommandService$1.service(JerseyContainerCommandService.java:149)
	at org.glassfish.admin.rest.adapter.RestAdapter.service(RestAdapter.java:159)
	at com.sun.enterprise.v3.services.impl.ContainerMapper$HttpHandlerCallable.call(ContainerMapper.java:429)
	at com.sun.enterprise.v3.services.impl.ContainerMapper.service(ContainerMapper.java:143)
	at org.glassfish.grizzly.http.server.HttpHandler.runService(HttpHandler.java:174)
	at org.glassfish.grizzly.http.server.HttpHandler.doHandle(HttpHandler.java:153)
	at org.glassfish.grizzly.http.server.HttpServerFilter.handleRead(HttpServerFilter.java:196)
	at org.glassfish.grizzly.filterchain.ExecutorResolver$9.execute(ExecutorResolver.java:88)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeFilter(DefaultFilterChain.java:246)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.executeChainPart(DefaultFilterChain.java:178)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.execute(DefaultFilterChain.java:118)
	at org.glassfish.grizzly.filterchain.DefaultFilterChain.process(DefaultFilterChain.java:96)
	at org.glassfish.grizzly.ProcessorExecutor.execute(ProcessorExecutor.java:51)
	at org.glassfish.grizzly.nio.transport.TCPNIOTransport.fireIOEvent(TCPNIOTransport.java:510)
	at org.glassfish.grizzly.strategies.AbstractIOStrategy.fireIOEvent(AbstractIOStrategy.java:82)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy.run0(WorkerThreadIOStrategy.java:83)
	at org.glassfish.grizzly.strategies.WorkerThreadIOStrategy$WorkerThreadRunnable.run(WorkerThreadIOStrategy.java:101)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.doWork(AbstractThreadPool.java:535)
	at org.glassfish.grizzly.threadpool.AbstractThreadPool$Worker.run(AbstractThreadPool.java:515)
	at java.base/java.lang.Thread.run(Thread.java:840)
]]
```

**Comment by avpinchuk** [comment]
Does this error persist for 7.1+?

* #25644

**Comment by pzygielo** [comment]
> Does this error persist for 7.1+?
> 
>     * Load `GlassFishRoleMapper` before installing the system-wide Exousia policy

I do not recollect to see this in latest 7.1s.
(On the other hand - I see this constantly in 6.2.5.)

My mistake: I was pretty sure it **was fixed** and that it happened in 7.0.x.

Thank you for linking PR above.
